library/src/main/java/org/xmtp/android/library/frames/FramesClient.kt

```diff
@@ -0,0 +1,92 @@
+package org.xmtp.android.library.frames
+
+import android.util.Base64
+import org.xmtp.android.library.Client
+import org.xmtp.android.library.frames.FramesConstants.PROTOCOL_VERSION
+import org.xmtp.android.library.messages.PrivateKeyBuilder
+import org.xmtp.android.library.messages.Signature
+import org.xmtp.android.library.messages.getPublicKeyBundle
+import org.xmtp.proto.message.contents.PublicKeyOuterClass.SignedPublicKeyBundle
+import java.security.MessageDigest
+import org.xmtp.proto.message.contents.Frames.FrameActionBody
+import org.xmtp.proto.message.contents.Frames.FrameAction
+
+class FramesClient(private val xmtpClient: Client, var proxy: OpenFramesProxy = OpenFramesProxy()) {
+
+    suspend fun signFrameAction(inputs: FrameActionInputs): FramePostPayload {
+        val opaqueConversationIdentifier = buildOpaqueIdentifier(inputs)
+        val frameUrl = inputs.frameUrl
+        val buttonIndex = inputs.buttonIndex
+        val inputText = inputs.inputText
+        val state = inputs.state
+        val now = System.currentTimeMillis() / 1000
+        val frameActionBuilder = FrameActionBody
+            .newBuilder()
+            .setFrameUrl(frameUrl)
+            .setButtonIndex(buttonIndex)
+            .setOpaqueConversationIdentifier(opaqueConversationIdentifier)
+            .setTimestamp(now)
+            .setUnixTimestamp(now.toInt())
+
+        if (inputText != null) {
+            frameActionBuilder.inputText = inputText
+        }
+        if (state != null) {
+            frameActionBuilder.state = state
+        }
+
+        val toSign = frameActionBuilder.build()
```

Comment:
This is a total fine way of doing this but in all the other places in the code we actually do it like this

```suggestion
        val toSign = FrameActionBody.newBuilder().also { frame ->
            frame.frameUrl = frameUrl
            frame.buttonIndex = buttonIndex
            frame.opaqueConversationIdentifier = opaqueConversationIdentifier
            frame.timestamp = now
            frame.unixTimestamp = now.toInt()
            frame.inputText  = inputText
            frame.state = state
        }.build()
```